packages/macro/test/js-useLingui.ts

```diff
@@ -0,0 +1,380 @@
+  const a = _t2(
+    /*i18n*/
+    {
+      id: "xeiujy",
+      message: "Text",
+    }
+  );
+}
+`,
+  },
+  {
+    name: "support nested macro",
+    input: `
+import { useLingui, plural } from '@lingui/macro';
+
+function MyComponent() {
+  const { t } = useLingui();
+  const a = t\`Text \${plural(users.length, {
+          offset: 1,
+          0: "No books",
+          1: "1 book",
+          other: "# books"
+        })}\`;
+}
+
+    `,
+    expected: `
+import { useLingui } from "@lingui/react";
+function MyComponent() {
+  const { _: _t } = useLingui();
+  const a = _t(
+    /*i18n*/
+    {
+      id: "hJRCh6",
+      message:
+        "Text {0, plural, offset:1 =0 {No books} =1 {1 book} other {# books}}",
+      values: {
+        0: users.length,
+      },
+    }
+  );
+}
+`,
+  },
+  {
+    name: "support message descriptor",
+    input: `
+import { useLingui } from '@lingui/macro';
+
+function MyComponent() {
+  const { t } = useLingui();
+  const a = t({ message: "Hello", context: "my custom" });
+}
+    `,
+    expected: `
+    import { useLingui } from "@lingui/react";
+    function MyComponent() {
+  const { _: _t } = useLingui();
+  const a = _t(
+    /*i18n*/
+    {
+      context: "my custom",
+      message: "Hello",
+      id: "BYqAaU",
+    }
+  );
+}`,
+  },
+  {
+    name: "support passing t variable as dependency",
+    input: `
+import { useLingui } from '@lingui/macro';
+
+function MyComponent() {
+  const { t } = useLingui();
+  const a = useMemo(() => t\`Text\`, [t]);
+}
+    `,
+    expected: `
+import { useLingui } from "@lingui/react";
+function MyComponent() {
+  const { _: _t } = useLingui();
+  const a = useMemo(
+    () =>
+      _t(
+        /*i18n*/
+        {
+          id: "xeiujy",
+          message: "Text",
+        }
+      ),
+    [_t]
+  );
+}
+`,
+  },
+  {
+    name: "transform to standard useLingui statement",
+    input: `
+import { useLingui } from '@lingui/macro';
+
+function MyComponent() {
+  const { i18n, t } = useLingui();
+
+  console.log(i18n);
+  const a = t\`Text\`;
+}
+    `,
+    expected: `
+import { useLingui } from "@lingui/react";
+function MyComponent() {
+  const { i18n, _: _t } = useLingui();
+  console.log(i18n);
+  const a = _t(
+    /*i18n*/
+    {
+      id: "xeiujy",
+      message: "Text",
+    }
+  );
+}
+`,
+  },
+  {
+    name: "work with existing useLingui statement",
+    input: `
+import { useLingui as useLinguiMacro } from '@lingui/macro';
+import { useLingui } from '@lingui/react';
+
+function MyComponent() {
+  const { _ } = useLingui();
+
+  console.log(_);
+  const { t } = useLinguiMacro();
+  const a = t\`Text\`;
+}
+    `,
+    expected: `
+import { useLingui } from "@lingui/react";
+function MyComponent() {
+  const { _ } = useLingui();
+  console.log(_);
+  const { _: _t } = useLingui();
+  const a = _t(
+    /*i18n*/
+    {
+      id: "xeiujy",
+      message: "Text",
+    }
+  );
+}
+`,
+  },
+
+  {
+    // todo: implement this
+    skip: true,
```

Comment:
just flagging this

Reply:
😬 thanks, i missed that one.

Reply:
Yes, i wrote about this case in the comments. I'm working on next PR which will fix it